storage/store/sql/sql.go

```diff
@@ -56,17 +54,34 @@ type Store struct {
 	// writeThroughCache is a cache used to drastically decrease read latency by pre-emptively
 	// caching writes as they happen. If nil, writes are not cached.
 	writeThroughCache *gocache.Cache
+
+	// maximumNumberOfResults is the maximum number of results that an endpoint can have
+	maximumNumberOfResults int
+	// maximumNumberOfEvents is the maximum number of events that an endpoint can have
+	maximumNumberOfEvents int
+
+	// eventsCleanUpThreshold is a maximum number of events before triggering a clean up
+	eventsCleanUpThreshold int
+	// resultsCleanUpThreshold is a maximum number of results before triggering a clean up
+	resultsCleanUpThreshold int
```

Comment:
Instead of defining `eventsCleanUpThreshold` and `resultsCleanUpThreshold` in the store, we should create constants named something like `eventsAboveMaximumCleanUpThreshold` and `resultsAboveMaximumCleanUpThreshold` & set both of these to `10`.

Then we can just compute the threshold based on `store.maximumNumberOfEvents + eventsAboveMaximumCleanUpThreshold` and `store.maximumNumberOfResults + resultsAboveMaximumCleanUpThreshold`.